docs/reference.rst

```diff
@@ -921,12 +921,40 @@ The following fields are supported for image sections:
 ``filename`` (required)
   Name of the image file (relative to bundle content).
 
+  .. note::
+    While RAUC can automatically detect the image type from filename extensions
+    for backward compatibility, it is recommended to explicitly specify the
+    ``type`` field for new configurations. The filename extension-based detection
+    is considered deprecated and may be removed in future versions.
+
   .. important::
     RAUC uses the file name extension and the slot type to decide how to
     extract the image file content to the slot.
     Make sure to only use :ref:`supported file name extensions
     <sec-ref-supported-image-types>`!
 
+``type`` (required)
+  Specifies the type of the image content. This determines how RAUC will process
+  the image during installation. If not specified, the type is automatically
+  detected based on the filename extension and target slot type.
+
+  Supported values are:
```

Comment:
We should refer to the "Supported Image Types" instead of duplicating the list here.